code/modules/atmospherics/machinery/pipes/smart.dm

```diff
@@ -12,69 +12,83 @@ GLOBAL_LIST_INIT(atmos_components, typecacheof(list(/obj/machinery/atmospherics)
 	pipe_state = "manifold4w"
 	///Current active connections
 	var/connections = NONE
+	///Was this pipe created during map load
+	var/map_loaded_pipe = FALSE
+
+/obj/machinery/atmospherics/pipe/smart/Initialize(mapload)
+	map_loaded_pipe = mapload
+	return ..()
+
+///helper function to append all directions into an single bit flag
+/obj/machinery/atmospherics/pipe/smart/proc/append_directions(list/spanning_directions)
+	var/bit_flag = 0
+	for(var/i in 1 to spanning_directions.len)
```

Comment:
Is spanning_directions null-safe (as in, we will never call this proc with this being null)? I would probably still just do `length(spanning_directions)` imo

Reply:
It is null safe as we only call it inside update_pipe_icon() with get_initial_directions()[which always returns a non null llst] as parameter.  But sure ill implement your suggestion